EDU-928: Redirect Cloud release notes (#2207)
temporalio/documentation: changelog/july-release-2022.md

:::

> - Temporal Cloud release notes are now available at `docs.temporal.io/cloud/release-notes` (they have since moved to [temporal.io/change-log/product-area/cloud](https://temporal.io/change-log/product-area/cloud)).

Reviewer: ```suggestion
- Temporal Cloud release notes are now available at `docs.temporal.io/cloud/release-notes`.
  (After this changelog was published, the release notes moved to [temporal.io/change-log/product-area/cloud](https://temporal.io/change-log/product-area/cloud).)
```
